webapp/src/ts/modals/edit-user/update-password.component.ts

```diff
@@ -73,12 +74,13 @@ export class UpdatePasswordComponent {
         }
       }
     } catch (error) {
-      if (error.status === 0) { // Offline status
+      const status = getProperty(error, 'status');
+      if (status === 0) { // Offline status
```

Comment:
I'm not sold on this change yet... I agree we shouldn't assume anything thrown/caught is an `Error` object and the `unknown` type is best for this purpose because it forces us to do proper validation in the `catch` clause. But this change barely gets us halfway there: it checks that the property exists but it doesn't help with type-safety.
I don't think a generic lib file can solve this problem on its own. I think this sort of validation should be limited to its respective `catch` clause. For example here this could look like:

```ts
import { HttpErrorResponse } from '@angular/common/http';

try {
	// ...
} catch {
	if (error instanceof HttpErrorResponse) {
		const status = error.status;
//			^? number
	}
}

Reply:
:raised_hands: This is just the kind of push back I was hoping for! I am very interested in digging in deeper here to find the best solution.  

The challenge I see with doing something like `error instanceof HttpErrorResponse` in existing code is that we would have to be _absolutely certain of what is in scope for the try-block and what kind of errors might be thrown._  Taking this code as a good example, there is quite a bit of logic wrapped in the try-block and I am not confident that _anything that was being thrown with a `status` property,_ is definitely going to be `instanceof HttpErrorResponse`.  For example, even with a local Pouch db, if you call `get` with a docId that does not exist, it will throw an error with `status: 404`, but I do not think it will be a `HttpErrorResponse`...

This was the main reason I settled on this half-baked pattern matching approach.  :grimacing: 

> his change barely gets us halfway there: it checks that the property exists but it doesn't help with type-safety.

100% agree.  But, I really think that if we are going to do nested type-checking, we should revisit the idea of adding a schema validation library.... :grimacing: I have found `effect/schema` to be super powerful in chtoolbox and I think we could reduce a lot of manual logic that we have by pulling in something like that.  (FTR, not saying we go with `effect/schema` here in cht-core.... :sweat_smile:  Zod, seems like the default option, but I find https://github.com/fabian-hiller/valibot to be very interesting as well, particularly with its claims of superior tree-shaking....)

Reply:
Did a simple side-by-side comparison of `zod` vs `valibot` for evaluating the schema of this error.  The code is nearly identical:

`zod`:

```ts
      const MyErrorSchema = z.object({
        status: z.number().optional(),
      });
      const myError = MyErrorSchema.parse(error);
```

`valibot`:

```ts
      const MyErrorSchema = v.object({
        status: v.optional(v.number()),
      });
      const myError = v.parse(MyErrorSchema, error);
```

Then, I used `npm run build` to build the prod assets (with tree-shaking) and compared the sizes of `api/build/static/webapp`:

- Current: `8384K`
- Zod: `8444K` - `+60K`
- Valibot: `8388K` - `+4K`

This is inline with [valibot's claims](https://valibot.dev/guides/comparison/) to be significantly more efficient than Zod because the structure of the project is optimized for tree shaking.

IMHO `+4K` is worth it for sane schema validation.  In reality, we should be able to replace a lot of janky manual code checks with valibot.